assets/components/countdown/countdown.jsx

```diff
@@ -0,0 +1,89 @@
+// @flow
+
+// ----- Imports ----- //
+import React, { Component } from 'react';
+
+
+// ---- Types ----- //
+type CountdownTime = {
+  t: number,
+  days: number,
+  hours: number,
+  minutes: number,
+  seconds: number,
+}
+
+type PropTypes = {
+  date: number[]
+};
+
+type StateTypes = {
+  countdown: CountdownTime
+}
+
+
+// ---- Helpers ----- //
+const addLeadingZeros = (value: number): string => {
+  let valueStr = String(value);
+  while (valueStr.length < 2) {
+    valueStr = `0${valueStr}`;
+  }
+  return valueStr;
+};
+
+const calculateCountdown = (endDate: number[]): CountdownTime => {
+  const t = new Date(...endDate).getTime() - Date.now();
+
+  const seconds = Math.floor((t / 1000) % 60);
+  const minutes = Math.floor((t / 1000 / 60) % 60);
+  const hours = Math.floor((t / (1000 * 60 * 60)) % 24);
+  const days = Math.floor(t / (1000 * 60 * 60 * 24));
+
+  return {
+    t, seconds, minutes, hours, days,
+  };
+};
+
+
+// ----- Component ----- //
+export default class Countdown extends Component<PropTypes, StateTypes> {
```

Comment:
We used to avoid stateful components, maybe @Ap0c has an updated point of view about this.